setup.py

```diff
@@ -4,9 +4,38 @@
 import codecs
 import os
 import re
+import sys
 
 from setuptools import setup
 
+CURRENT_PYTHON = sys.version_info[:2]
+REQUIRED_PYTHON = (3, 6)
+
+# This check and everything above must remain compatible with Python 2.7.
+if CURRENT_PYTHON < REQUIRED_PYTHON:
+    sys.stderr.write(
+        """
+==========================
+Unsupported Python version
+==========================
+This version of urllib3 requires Python {}.{}, but you're trying to
+install it on Python {}.{}.
+This may be because you are using a version of pip that doesn't
+understand the python_requires classifier. Make sure you
+have pip >= 9.0 and setuptools >= 24.2, then try again:
+    $ python -m pip install --upgrade pip setuptools
+    $ python -m pip install urllib3
+This will install the latest version of urllib3 which works on your
+version of Python. If you can't upgrade your pip (or Python), request
+an older version of urllib3:
+    $ python -m pip install "urllib3<2"
+""".format(
+            *(REQUIRED_PYTHON + CURRENT_PYTHON)
```

Comment:
This seems to be a bit of a hack to work around pyupgrade f-strings, it might be easier to hardcode these. Like the pip and setuptools versions

Reply:
This constant is only used here, but the same value is hardcoded later in python_requires:

https://github.com/urllib3/urllib3/pull/2070/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R112

Reply:
pyupgrade or f-strings did not exist when this was written in 2016. I can hardcode 3.6, but `CURRENT_PYTHON` will stay dynamic

Reply:
`REQUIRED_PYTHON` is used above, so I would not gain much by inlining it. Sorry, given the two other approvals I decided against doing this